File: core/src/main/java/dev/bee/kanjianki/core/StudyTokenPolicy.java
package dev.bee.kanjianki.core;

import java.util.UUID;

public final class StudyTokenPolicy {
    private StudyTokenPolicy() {
    }

    public static String studyItem(String kanji, String activeToken) {
        return existingOrNew(activeToken, kanji + "-");
    }

    private static String existingOrNew(String activeToken, String prefix) {
        if (activeToken != null && !activeToken.isEmpty()) {
            return activeToken;
        }
        return prefix + UUID.randomUUID();
    }
}

File: core/src/main/kotlin/dev/bee/kanjianki/core/StudyTokenPolicy.kt
package dev.bee.kanjianki.core

import java.util.UUID

object StudyTokenPolicy {
    @JvmStatic
    fun studyItem(kanji: String?, activeToken: String?): String {
        return existingOrNew(activeToken, "$kanji-")
    }

    private fun existingOrNew(activeToken: String?, prefix: String): String {
        if (!activeToken.isNullOrEmpty()) {
            return activeToken
        }
        return prefix + UUID.randomUUID()
    }
}
